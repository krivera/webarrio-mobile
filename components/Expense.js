import React from 'react';
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  View
} from 'react-native';
import Collapsible from 'react-native-collapsible';
import { Ionicons, Foundation, SimpleLineIcons } from '@expo/vector-icons';
import { withGlobalize } from 'react-native-globalize';
import WebarrioIcon from '../components/WebarrioIcon';
import Colors from '../constants/Colors';
import { MonthsFull, ExpenseDetails } from '../constants/utils';

class Expense extends React.Component{
  constructor(props) {
    super(props);
  
    this.state = {
      collapsed: true
    };
  }

  toggleCollapse = () => {
    this.setState({
      collapsed: !this.state.collapsed
    });
  }

  render(){
    const { expense, globalize } = this.props;
    const formatter = globalize.getNumberFormatter({});
    let checkmark = 'ios-checkmark-circle';
    if(!this.state.collapsed){
      checkmark += '-outline'
    }
    return (
      <View style={styles.expense}>
        <TouchableOpacity
          onPress={this.toggleCollapse}
          style={styles.header}
        >
          <Ionicons
            size={25}
            name={checkmark}
            color={Colors.orange}
            style={styles.collapseButton}
          /> 
          <Text>{MonthsFull[parseInt(expense.month) - 1]}</Text>
          <Text style={[styles.number, styles.total]}>
            $ {formatter(expense.total)}
          </Text>
        </TouchableOpacity>
        <Collapsible collapsed={this.state.collapsed}>
          <View style={styles.detail}>
            <Text style={styles.detailHeader}>Detalle de gastos</Text>
            {ExpenseDetails.map(detail => (
              <View style={styles.row} key={detail.key}>
                <View style={styles.iContainer}>
                  {detail.icon}
                </View>
                <Text>{detail.label}</Text>
                <Text style={styles.number}>
                  $ {formatter(expense[`total_${detail.key}`])}
                </Text>
              </View>
            ))}
          </View>
        </Collapsible>
      </View>
    );
  }
}

export default withGlobalize(Expense);

export const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    flex: 1,
    padding: 15,
    alignItems: 'center'
  },
  number: {
    flex: 1,
    textAlign: 'right',
  },
  total: {
    color: Colors.border,
    fontSize: 22
  },
  row: {
    flexDirection: 'row',
    alignItems: 'center',
    width: 200
  },
  iContainer: {
    width: 30, 
    alignItems: 'center'
  },
  detail: {
    alignSelf: 'center',
    paddingBottom: 15
  },
  detailHeader: {
    fontWeight: 'bold'
  },
  collapseButton: {
    paddingRight: 10
  },
  expense: {
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderBottomColor: Colors.border
  }
});
