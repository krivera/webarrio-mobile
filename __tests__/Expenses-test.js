import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Expense from '../components/Expense';
import { FormattedProvider } from 'react-native-globalize';
import AddExpenseScreen from '../screens/AddExpense';

it('renders individual expense', () => {
  const expense = {
    total: 123123,
    month: 'Enero',
    year: 2018,
    total_electricity: 123,
    total_water: 124,
    total_staff: 125,
    total_gas: 126,
    total_extras: 127,
  }
  const tree = renderer.create(
    <FormattedProvider locale='es'>
      <Expense expense={expense} />
    </FormattedProvider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders expenses form', () => {
  const tree = renderer.create(<AddExpenseScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
