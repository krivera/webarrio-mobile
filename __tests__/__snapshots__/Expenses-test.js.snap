// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders expenses form 1`] = `
<View>
  <View>
    <View
      style={undefined}
    >
      <RCTPicker
        items={
          Array [
            Object {
              "label": "ENERO",
              "textColor": undefined,
              "value": 1,
            },
            Object {
              "label": "FEBRERO",
              "textColor": undefined,
              "value": 2,
            },
            Object {
              "label": "MARZO",
              "textColor": undefined,
              "value": 3,
            },
            Object {
              "label": "ABRIL",
              "textColor": undefined,
              "value": 4,
            },
            Object {
              "label": "MAYO",
              "textColor": undefined,
              "value": 5,
            },
            Object {
              "label": "JUNIO",
              "textColor": undefined,
              "value": 6,
            },
            Object {
              "label": "JULIO",
              "textColor": undefined,
              "value": 7,
            },
            Object {
              "label": "AGOSTO",
              "textColor": undefined,
              "value": 8,
            },
            Object {
              "label": "SEPTIEMBRE",
              "textColor": undefined,
              "value": 9,
            },
            Object {
              "label": "OCTUBRE",
              "textColor": undefined,
              "value": 10,
            },
            Object {
              "label": "NOVIEMBRE",
              "textColor": undefined,
              "value": 11,
            },
            Object {
              "label": "DICIEMBRE",
              "textColor": undefined,
              "value": 12,
            },
          ]
        }
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={3}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={undefined}
    >
      <RCTPicker
        items={Array []}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  )}
</View>
`;

exports[`renders individual expense 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#92a2a2",
      "borderBottomWidth": 0.5,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "opacity": 1,
        "padding": 15,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "flex": 1,
            "textAlign": "right",
          },
          Object {
            "color": "#92a2a2",
            "fontSize": 22,
          },
        ]
      }
    >
      $ 
      123.123
    </Text>
  </View>
  <View
    collapsable={undefined}
    pointerEvents="none"
    style={
      Object {
        "height": 0,
        "overflow": "hidden",
      }
    }
  >
    <View
      collapsable={undefined}
      onLayout={[Function]}
      style={Object {}}
    >
      <View
        style={
          Object {
            "height": null,
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "paddingBottom": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Detalle de gastos
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "width": 200,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "width": 30,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              LUZ
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "flex": 1,
                  "textAlign": "right",
                }
              }
            >
              $ 
              123
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "width": 200,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "width": 30,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              AGUA
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "flex": 1,
                  "textAlign": "right",
                }
              }
            >
              $ 
              124
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "width": 200,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "width": 30,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              GAS
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "flex": 1,
                  "textAlign": "right",
                }
              }
            >
              $ 
              126
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "width": 200,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "width": 30,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              PERSONAL
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "flex": 1,
                  "textAlign": "right",
                }
              }
            >
              $ 
              125
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "width": 200,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "width": 30,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              OTROS
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "flex": 1,
                  "textAlign": "right",
                }
              }
            >
              $ 
              127
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
